### ISSUE TITLE
Dependency checking: The RMblast engine is not installed in RepeatMasker!

### ISSUE THREAD

**Issue by Donbbit** [oushujun/LTR_retriever#43]
I use conda to install LTR_retriever ：
conda create -n LTR_retriever
source activate LTR_retriever
conda install -c conda-forge perl perl-text-soundex
conda install -c bioconda cd-hit repeatmasker
git clone https://github.com/oushujun/LTR_retriever.git
./LTR_retriever/LTR_retriever -h
 
But when I run it with : LTR_retriever -genome Athaliana_167_TAIR9.fa -infinder Athaliana_167_TAIR9.fa.LTR.fa -verbose 
the error occurs.

**Comment by oushujun** [comment]
Hello @Donbbit ,

Thank you for using LTR_retriever. Have you solved this issue? If so, could you post how you solve it? Other users may benefit from your experience, thanks!

Best,
Shujun

**Comment by nm100** [comment]
Hi @Donbbit, 

Not sure if you were able to solve this issue, but here's a quick solution to this problem:

Try to run RepeatMasker independently with the following command first:
RepeatMasker -e ncbi -q -pa 1 -no_is -norna -nolow dummy060817.fa.$rand -lib dummy060817.fa.$rand

if you are getting the following error on running the above command:
Taxononmy::new() Could not locate the taxonomy data file!

Then, you need to go to the RepeatMasker lib directory and unzip the taxonomy.data.gz file.

This will solve the RepeatMasker dependency issue.

Thanks!

**Comment by oushujun** [comment]
@nm100 Thanks Nancy!

The RepeatMasker lib directory is located here:
`$path/miniconda2/envs/LTR_retriever/share/RepeatMasker/Libraries/`, where $path is the installation path of your conda.

Best,
Shujun

**Comment by moontree1985** [comment]
Hi, Shujun,

I am facing the same problem with dependency of RMblast engine. I tried to follow Nancy's suggestions. I don't have Taxononmy::new() error but successfully got the following files:
dummy060817.fa.926962      dummy060817.fa.926962.masked  dummy060817.fa.926962.nin  dummy060817.fa.926962.ori.out  dummy060817.fa.926962.tbl
dummy060817.fa.926962.cat  dummy060817.fa.926962.nhr     dummy060817.fa.926962.nsq  dummy060817.fa.926962.out

However, when I run ltrretriever 2.5, I still got error:
Dependency checking: The RMblast engine is not installed in RepeatMasker!

I don't have this error when I use ltrretriever 20170514. However, the problem of ltrretriever 20170514 is I can not get LAI result. I need to run LAI separately to get LAI result.

Best,

Ying

**Comment by oushujun** [comment]
Summary for the fix and with more details:

1. If you install dependencies via conda and encounter this error in LTR_retriever:
 >Dependency checking: The RMblast engine is not installed in RepeatMasker!

2. Your RepeatMasker is likely improperly installed. To test out, simply run this command in your LTR_retriever working directory:
 `RepeatMasker -e ncbi -q -pa 1 -no_is -norna -nolow dummy060817.fa* -lib dummy060817.fa*`

3. If you enounter this error:
 >RepeatMasker version development-$Id: RepeatMasker,v 1.332 2017/04/17 19:01:11 rhubley Exp $
Search Engine: NCBI/RMBLAST [ 2.6.0+ ]
Taxononmy::new() **Could not locate the taxonomy data file!**
 at /home/oushujun/las/bin/miniconda2/envs/LTR_retriever/share/RepeatMasker/RepeatMasker line 638.

Then yes, it confirms that your RepeatMasker is improperly installed.

4. To fix RepeatMasker, list this directory:
`ls -l $path/miniconda2/envs/LTR_retriever/share/RepeatMasker/Libraries/`
 where `$path` is the installation path of your conda.
You should see this file: `taxonomy.dat.gz`

5. Do these three steps (remember to replace $path with your actual path). It just simply copies it out, unzip it, and move the unzipped version back:
`cp $path/miniconda2/envs/LTR_retriever/share/RepeatMasker/Libraries/taxonomy.dat.gz ./`
`gunzip taxonomy.dat.gz`
`mv taxonomy.dat $path/miniconda2/envs/LTR_retriever/share/RepeatMasker/Libraries/`

Your RepeatMasker should be fixed. To test out:

6. Rerun:  `$ RepeatMasker -e ncbi -q -pa 1 -no_is -norna -nolow dummy060817.fa* -lib dummy060817.fa*`
You should see something like:
>RepeatMasker version development-$Id: RepeatMasker,v 1.332 2017/04/17 19:01:11 rhubley Exp $
Search Engine: NCBI/RMBLAST [ 2.6.0+ ]
Master RepeatMasker Database: /home/oushujun/las/bin/miniconda2/envs/LTR_retriever/share/RepeatMasker/Libraries/RepeatMaskerLib.embl ( Complete Database: dc20181026 )
Custom Repeat Library: dummy060817.fa.818966
Warning...unknown stuff 
Building general libraries in: /home/oushujun/las/bin/miniconda2/envs/LTR_retriever/share/RepeatMasker/Libraries/dc20181026/general
analyzing file dummy060817.fa.818966
identifying matches to dummy060817.fa.818966 sequences in batch 1 of 1
processing output: 
cycle 1 
cycle 2 
Generating output... 
masking
**done**

**Comment by aaronphillips7493** [comment]
Hello,

I am getting the same error about RMblast engine when I try to run LTR_retriever.

When I try to run:
`RepeatMasker -e ncbi -q -pa 1 -no_is -norna -nolow dummy060817.fa* -lib dummy060817.fa*`

I get the error:
`RepeatMasker can not handle filenames with the special character "*" as in the file "dummy060817.fa*"`

Is there a fix for this?

Thank you :)

**Comment by oushujun** [comment]
@aaronphillips7493 "*" is the file extension of your dummy file. LTR_retriever will add the system date as an extension to avoid corruption. You may copy the dummy file from the LTR_retriever folder to another place for this test. Or you may simply reinstall RepeatMasker from [conda](https://anaconda.org/bioconda/repeatmasker).

**Comment by aaronphillips7493** [comment]
Hey,

I am trying to run this code:
`LTR_retriever/LTR_retriever -genome ragtag_nuclear_assemblyhypo_polishingragoo_scaffolds/ragtag.scaffolds.fasta -inharvest ragtag_nuclear_assemblyhypo_polishingragoo_scaffolds/ragtag.scaffolds.fasta.rawLTR.scn -threads 10`

Now when I try to run LTR_Retriever I get the error:
`Dependency checking: Error: The RMblast engine is not installed in RepeatMasker!`

I have tried re-installing everything as per the online instructions, and followed the steps above to resolve the issue. 

When I run:
`RepeatMasker -e ncbi -q -pa 1 -no_is -norna -nolow dummy060817.fa* -lib dummy060817.fa*`

**Comment by oushujun** [comment]
@aaronphillips7493 

You may need to reinstall RMblast as suggested in this thread #15 by @weihankk:

> My LTR_retriever version is 2.9.0, my RepeatMasker version is 4.0.9, and my rmblast version is 2.2.28. So I try to reinstall rmblast in another version by conda:
> 
> `conda install rmblast=2.6.0`
> 
> Fortunately, the error was resolved. I run LTR_retriever successfully!

**Comment by ElkanChang** [comment]
Hi Shujun,

I'm not using a conda version but I still met this problem with LTR_retriever-2.9.0.

However, I'm pretty sure my RepeatMasker is properly installed, and when I try the protocol you mentioned above:
```
RepeatMasker -e ncbi -q -pa 1 -no_is -norna -nolow dummy060817.fa* -lib dummy060817.fa*
```
Everything thing is normal, 
```
RepeatMasker -e ncbi -q -pa 1 -no_is -norna -nolow dummy060817.fa.458191 -lib dummy060817.fa.458191 
RepeatMasker version 4.1.1
Search Engine: NCBI/RMBLAST [ 2.10.0+ ]
Using Custom Repeat Library: dummy060817.fa.458191


analyzing file dummy060817.fa.458191
identifying matches to dummy060817.fa.458191 sequences in batch 1 of 1
processing output: 
cycle 1 
cycle 2 
Generating output... 
masking
done
```
It just didn't work within the LTR_retriever, can you help me with this ?

BRs

**Comment by oushujun** [comment]
Hi BRs,

If you have a working RepeatMasker, you can provid its path to
LTR_retriever. Please check -help for more instructions.

Best,
Shujun

**Comment by ElkanChang** [comment]
Hi Shujun,

I have tried the `-repeatmasker ` parameter and still got the same error...
```
##########################
### LTR_retriever v2.9.0 ###
##########################

Contributors: Shujun Ou, Ning Jiang

For LTR_retriever, please cite:

	Ou S and Jiang N (2018). LTR_retriever: A Highly Accurate and Sensitive Program for Identification of Long Terminal Repeat Retrotransposons. Plant Physiol. 176(2): 1410-1422.

For LAI, please cite:

	Ou S, Chen J, Jiang N (2018). Assessing genome assembly quality using the LTR Assembly Index (LAI). Nucleic Acids Res. 2018;46(21):e126.

Parameters: -genome genomic.fna -infinder finder.scn -inharvest harvest.scn -repeatmasker /home/elkan/MyFiles/software/RepeatMasker


Thu Apr  1 18:32:33 CEST 2021	Dependency checking: Error: The RMblast engine is not installed in RepeatMasker!
```
BRs
Elkan

**Comment by oushujun** [comment]
Hi Elkan,

Sorry for the delayed reply. It seems that the RepeatMasker you were providing did not have the RMblast engine deployed. You may want to reconfigure the program to use the RMblast engine.

Best,
Shujun

**Comment by zmz1988** [comment]
Hi all, I also got into this error that RMblast engine is not installed in RepeatMasker, after several months I made a successful run with LTR_retriever. I guess the new RepeatMasker does not include RMBlast by default. It took me a bit to find out how to re-configure the RepeatMasker in miniconda environment, and would like to share here, in case anyone needs it.

First thing is to `conda install rmblast`, and then in the environment that repeatMasker was build find the path where contain the configure file of RepeatMasker (e.g. /miniconda3/envs/EDTA/share/RepeatMasker/configure ), then run `perl ./configure` and follow the instruction. Some info can be found here https://supcom.hgc.jp/english/utili_info/manual/RepeatMasker.html
